### PR TITLE
Update name of DataFrame.WriteCsv to SaveCsv

### PR DESCRIPTION
**Pull request by dakersnar**
It seems like there was [general agreement](https://github.com/dotnet/machinelearning/issues/5883) to update the name about a year ago, but it has been a while, so I want to resurface this and ask the following questions:

1. Do we still want to go ahead with this change?
2. Is this considered a "breaking change"? If yes, are there any repercussions we should be aware of?
3. Is there anything else I'm missing that might need to be updated (documentation, other files, etc)?

### CODE REVIEW

**Comment by eerhardt** [comment]
> Is this considered a "breaking change"? If yes, are there any repercussions we should be aware of?

Yes, it is a breaking change, but we haven't shipped the package as "stable" yet. Maybe we could add `[Oboslete]` to WriteCsv for one preview release, and then remove the obsolete API in the next release?

**Comment by dakersnar** [comment]
@eerhardt Good idea, I just adding a commit doing that.

File: src/Microsoft.Data.Analysis/DataFrame.IO.cs [hunk 1]
```diff
@@ -396,6 +396,7 @@ public TextReader GetTextReader()
             return ReadCsvLinesIntoDataFrame(wrappedStreamReaderOrStringReader, separator, header, columnNames, dataTypes, numberOfRowsToRead, guessRows, addIndexColumn);
         }
 
+
```
Comment (dakersnar) [inline src/Microsoft.Data.Analysis/DataFrame.IO.cs:399]:
```suggestion
```

File: src/Microsoft.Data.Analysis/DataFrame.IO.cs [hunk 2]
```diff
@@ -405,30 +405,65 @@ public TextReader GetTextReader()
...
             using (FileStream csvStream = new FileStream(path, FileMode.Create))
             {
-                WriteCsv(dataFrame: dataFrame, csvStream: csvStream,
+                SaveCsv(dataFrame: dataFrame, csvStream: csvStream,
                            separator: separator, header: header,
                            encoding: encoding, cultureInfo: cultureInfo);
             }
         }
 
+
```
Comment (dakersnar) [inline src/Microsoft.Data.Analysis/DataFrame.IO.cs:437]:
```suggestion
```

File: src/Microsoft.Data.Analysis/DataFrame.IO.cs [hunk 3]
```diff
@@ -405,30 +405,64 @@ public TextReader GetTextReader()
...
-                WriteCsv(dataFrame: dataFrame, csvStream: csvStream,
+                SaveCsv(dataFrame: dataFrame, csvStream: csvStream,
                            separator: separator, header: header,
                            encoding: encoding, cultureInfo: cultureInfo);
             }
         }
 
         /// <summary>
-        /// Writes a DataFrame into a CSV.
+        /// Saves a DataFrame into a CSV.
```
Comment (dakersnar) [inline src/Microsoft.Data.Analysis/DataFrame.IO.cs:438]:
```suggestion
        /// Writes a DataFrame into a CSV.
```

File: src/Microsoft.Data.Analysis/DataFrame.IO.cs [hunk 4]
```diff
@@ -405,13 +405,30 @@ public TextReader GetTextReader()
         /// <param name="header">has a header or not</param>
         /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
         /// <param name="cultureInfo">culture info for formatting values</param>
+        [Obsolete("Use SaveCsv")]
```
Comment (eerhardt) [inline src/Microsoft.Data.Analysis/DataFrame.IO.cs:408]:
```suggestion
        [Obsolete("WriteCsv is obsolete and will be removed in a future version. Use SaveCsv instead.")]
```

**Review by eerhardt — APPROVED** [review]
LGTM - just a minor suggestion on the message wording.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**